### DISCUSSION TITLE
Problem updating information, sites with Joomla3 and Connector 1.0.2

### DISCUSSION THREAD

**Discussion by nunoleite** [akeeba/panopticon#389] [Q&A (Community Support)]
Hi,

I have installed Connector 1.0.2 on some Joomla3 sites, and after that i can't update site information.

I'm having these Errors:
```
An error occurred loading the site extensions information.
Server error: `GET https://www.xxxxx.pt/panopticon_api/index.php/v1/panopticon/extensions?page%5Blimit%5D=10000` resulted in a `500 Internal Server Error` response
```
```
An error occurred updating the site information.
Server error: `GET https://www.xxxxx.pt/panopticon_api/index.php/v1/panopticon/core/update?force=1` resulted in a `500 Internal Server Error` response
```

Sites with Joomla 4 and with connector 1.0.2 doesn't show the error and appear to be working.
And sites with Joomla 3 and Connector 1.0.1 also doesn't show the error.

As i can see the ones that are throwing the error can't detect the connector 1.0.2, because they still show the info that connector is 1.0.1.
```
Panopticon Connector v.1.0.1
API level 100
```
So, something is wrong and i can't figure out the problem. I already tried to disable some extensions in these sites, but can't resolve.

Reinstalling Connector 1.0.1 removes the error, but for example when searching for updates extensions it can't find the new connector 1.0.2

Can someone help finding the problem?

Note: Is there a possibility that another update site from another extension is breaking this?

**Comment by nikosdion** [comment]
The only changes from 1.0.1 to 1.0.2 are:
- Collect server information.
- Tell Joomla! to refresh its updates cache when requesting update information.

Neither of these are Joomla version-specific. 

However, you said something important:

> As i can see the ones that are throwing the error can't detect the connector 1.0.2, because they still show the info that connector is 1.0.1.

This means that either the update failed **OR** you are still running version 1.0.1 because these sites cannot see and/or install updates. This explains why they fail to work correctly. If Joomla's extensions update does not work for any reason on your site you cannot connect them to Panopticon since retrieving update information (for extensions and Joomla! itself; both are collected by the same Joomla! core code) fails.

**Comment by nunoleite** [comment]
I have searched and searched, and even disabled lot's of extensions but can't find the source of the problem.
Some extensions do show updates available, and i can update, but others don't...
For example Panopticon Connector doesn't show updates available.
I really can't find the problem, I don't know if it's the Connector or the website, or even if it's the Panopticon monitoring server.

**Reply by nikosdion** [reply]
I can answer the last question easily. Do you see the update to the connector when logged into the backend of your site (NOT Panopticon)? If you don't, it's a problem with your site either missing the update site for the connector, or not being able to connect to the update server. If you see it, then your site does not connect with Panopticon.

Remember, Panopticon does not look for updates itself. It shouldn't. Updates in Joomla! can have restrictions on the Joomla! version range (minimum and maximum), minimum PHP version, database type, and minimum database server version. Therefore, which is the “latest” version of an extension is a question only Joomla! running on the site itself can answer. For example, you can't run Akeeba Backup 9.8 on any Joomla! 3 site, or any Joomla! 4 site on PHP 7.3. Moreover, the format Joomla! understands about these restrictions _does_ change over time, which means that trying to create a parser for update sites' contents in Panopticon would result in the results Panopticon detects about updates and what Joomla! detects to differ. The correct results are those identified by Joomla!, which is why Panopticon always gets the update information from your site, not the update server of each individual extension.

**Reply by nunoleite** [reply]
I understand the second paragraph.

Regarding the first, i will assume that the problem is with the websites... I have some websites in different servers, but all with almost the same configurations.

> Do you see the update to the connector when logged into the backend of your site (NOT Panopticon)?

No, the Panopticon Connector can't find the new versions, but other extensions can.

What is strange it that i have a Joomla 4 that is working fine...

I will try to install a clean Joomla 3 and see what happens, and even another instance of Panopticon monitoring to see.

**Comment by nunoleite** [comment]
I have installed a clean joomla 3.10.12 site with php 7.4
Then installed Panopticon Connector for J3 1.0.1
Then i go to updates to search for the Connector 1.0.2, and nothing...
It should find the new Connector 1.0.2 to update, but no.
When i open the link https://akeeba-cdn.s3.amazonaws.com/updates/plg_system_panopticon.xml it shows the "<version>1.0.2</version>"
It seems to be another problem.

**Comment by nikosdion** [comment]
Is the update site for the extension published?

When you try to clear the update cache and fetch the updates again, does it complain about the update site for the plugin at all?

**Reply by nunoleite** [reply]
No, nothing, no updates, no warnings.
![image](https://github.com/akeeba/panopticon/assets/173281/dee98d48-e748-4411-9aa0-e59504da2b58)
![image](https://github.com/akeeba/panopticon/assets/173281/3c3964fa-8f28-4e32-a37c-6eaa5095d667)

**Comment by nunoleite** [comment]
Hi!,
I have investigated further, and i see an error on error_log when the Update site from Panopticon Connector is enabled.
`PHP Warning:  preg_match(): Compilation failed: missing closing parenthesis at offset 11 in /home/xxxxxx/public_html/libraries/src/Updater/Adapter/ExtensionAdapter.php on line 133`

I look into the xml file from the update site and i think the error is from here:
`<targetplatform name="joomla" version="(3\.(9|10)"/>`
this should be
`<targetplatform name="joomla" version="(3\.(9|10))"/>`

Can you confirm this?

Thanks.

**Reply by nikosdion** [reply]
Correct, this is the problem. I am replacing the XML on the update server. It will take about 10' for the change to make it to the CDN.

**Comment by nunoleite** [comment]
I see that the xml is corrected, but when i try to find for updates to Panopticon Connector 1.0.1 still can't find the update to 1.0.2.
This is very strange.
And this is happening with a completely new Joomla 3.10.12 website.
So i wander... is this just me?
Who else is having this problem?

**Reply by nikosdion** [reply]
Give it another try now. The `element` was wrong.

**Comment by nunoleite** [comment]
Great, that solves the problem of detecting the newer versions of Connector.

Now that the websites are updating automatically to connector 1.0.2 i have the problem that originally i have posted.

As the connectors are being updated now i have the problem that the Panopticon Server doesn't detect the version 1.0.2 of the Connector and maintains the info that they are version 1.0.1 and API level 100.
And they report error 500 when cheking for Joomla Updates and Extension Updates.

What can I do to find out the problem?

**Reply by nikosdion** [reply]
I can help you troubleshoot your issue, but I need the following connection information to your site.

**Site connection information**

* Login URL e.g. `https://www.example.com/administrator/index.php?secret123`
* Super User username
* Super User password

**(S)FTP connection information**

* Hostname
* Port (if other than 21 or 22)
* Username
* Password

Do **NOT** paste this information unencrypted. GitHub issues and discussions are publicly visible. Instead, follow the instructions below to send me an encrypted copy of the information which can only be decrypted by me:

* Go to https://keybase.io/encrypt
* In the **Recipient** field type `nikosdion`.
* It will show you my face, my name, and the fact that I have linked my same-named GitHub account; verify this information and click on my face.
* In the **Message to encrypt** area enter the information I asked you above.
* Click the Encrypt button.
* After a short while the text area changes its title to **The secret message**.
* Click in the text area to copy the secret (encrypted) message.
* Paste it to your next reply to me.

This is perfectly safe to do! It uses public key encryption, just like HTTPS, with a _much, MUCH stronger cipher_ (3744-bit Elliptic Curve Cryptography which is insanely secure). The message can only be decrypted with my secret key which is safely stored on a hardware token, protected by a PIN.

Please make sure that you test the access credentials using a separate browser / (S)FTP software than what you normally use to ensure that they really work. Please make sure that there are no geographic or other restrictions which may prevent me from accessing your site. Kindly note that I am on the [Asia/Nicosia timezone](https://www.timeanddate.com/worldclock/cyprus/nicosia) and I am handling support free of charge; some delays are to be expected.

**Comment by nunoleite** [comment]
Here it is.
Thank you for your support.

-----BEGIN PGP MESSAGE-----
Version: Keybase OpenPGP v2.1.13
Comment: https://keybase.io/crypto

wcBMA6+1LLjxrhABAQf/f7/8vC1XV6fUuRy/FENXLellOpZBzPh1ZpN5SEBz2w/T
fitxR+M47FS2NKYD08YuaDo6stVadZUnnqfPRjwcX150l50rYqLXDDQePMiMHI2h
xRL/iihFJffx2a8lDYeTiEKEXw6t4JDJeqd5PLyvfk3+7YpbvSxdd74LGARgrlH/
1F3Z/KPHl7tgICG0PwbX/5gBXXZyyGayRLI4vQgc4Exlly9dUo7F8FIpogt4AMxk
UFu9ZwyDpCQeOKkJwYPcBylB7P1J3SkDggC5/k1vn6hBg7f6jNSX58/b9u7ksHFl
/Fyn9mDGI/mXrz6XpH0dHVFS9H1bOCFY8gZUlkf45dLAEwEXjqo04xhHaRl5E0IV
oUr24/3CWGAOLUvMEzMpvAoTPDEf2VMNIppKzgAv2iaSFyKkXw0ZL+wf6I155QnN
LI4FjpPQuE1qKAhDS4Y+alksmE+Xuw37jkBPLge1UwuYInwNb9EMdEUiZmYUMgtM
EOKfNhvNhyc3yGMzQAvFlcQieeCgs2HNDRKjDDFNu8rRqKej+7JoBjCdxAyBWDKf
zUcmCsHwbVBavP6b8TCwFEBeVprCHFzN8/jXNrVvKdkXY0oXL4bghpPEhkDMXuoT
3uImSkg=
=FXDK
-----END PGP MESSAGE-----

**Reply by nikosdion** [reply]
The problem was that a required file (polyfills.php) was missing from the XML manifest. Please install version 1.0.3 which has just been released.

**Comment by nunoleite** [comment]
Hi!

I have installed on 1 website and it solved the detection of Connector, now i have:
Panopticon Connector v.1.0.3
API level 100

The Joomla update works too, and the PHP version too...

But the extensions now have a new error.

Here is the screenshot:
![Image 004](https://github.com/akeeba/panopticon/assets/173281/5bf17c5c-1356-4733-9cda-ae53208645e7)

And the error is:
Invalid (non-JSON) data returned from the Joomla! API. Probably a third party plugin is breaking the API application? Raw data as follows: {"errors":[{"title":"Argument 1 passed to Akeeba\\PanopticonConnector\\Model\\ExtensionsModel::refreshUpdateInformation() must be an instance of Akeeba\\PanopticonConnector\\Model\\mixed, bool given, called in \/home\/xxxxx\/public_html\/plugins\/system\/panopticon\/src\/Model\/ExtensionsModel.php on line 106"}]} 

Sorry to be so insistent, but i think i'm helping you solving some bugs.

Thanks.

**Comment by nunoleite** [comment]
When i add a new site with Connector J3 version 1.0.3 i have the same error and can't connect.
{"errors":[{"title":"Argument 1 passed to Akeeba\\PanopticonConnector\\Model\\ExtensionsModel::refreshUpdateInformation() must be an instance of Akeeba\\PanopticonConnector\\Model\\mixed, bool given, called in \/home\/xxxxx\/public_html\/plugins\/system\/panopticon\/src\/Model\/ExtensionsModel.php on line 106"}]}

**Reply by nikosdion** [reply]
I suppose that site is on PHP 7?

**Reply by nunoleite** [reply]
yes, 7.4

**Reply by nikosdion** [reply]
This will be fixed in the next release.

**Reply by nunoleite** [reply]
Thanks
